src/CacheManager.Core/Internal/CacheKeysHelper.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace CacheManager.Core.Internal
+{
+    /// <summary>
+    /// Helper functions for manipulating keys.
+    /// </summary>
+    static public class CacheKeysHelper
```

Comment:
Putting it in BaseCacheHandle works, and is only a little bit uglier (it can't be an extension method there because BaseCacheHandle is generic)

Keys.Where(condition).FilterBy(pattern)
vs.
Keys.Where(condition).Where(k => FilterBy(k,pattern))

Though it needs a different name when not an extension method.